### ISSUE TITLE
Really like this library

### ISSUE THREAD

**Issue by chrisnicola** [thebearingedge/pg-bump#46]
I've been spending all day looking for a migration solution that makes sense and this is the first one that really seemed to fit the bill.

I like that it is just raw SQL. However in discussing this with the team the one piece we are kind of unsure of is combining the up/down migrations in one file. In our minds this makes that file far less useful as it can't really be treated as a proper migration without parsing it first. I realize in the context of the tool this probably doesn't matter, but I was wondering have you thought about keeping the up/down migrations as separate files and what was the reason for keeping them together?

**Comment by thebearingedge** [comment]
Hello, Chris and thanks for reaching out. I too searched for a while and tried a few different packages before deciding to write `pg-bump`.

>  I was wondering have you thought about keeping the up/down migrations as separate files and what was the reason for keeping them together?

The majority of implementations that I referenced kept `up` and `down` commands collocated in one file ([K'nex](http://knexjs.org/#Migrations), [Active Record](http://edgeguides.rubyonrails.org/active_record_migrations.html#migration-overview), [SQL Migrate for Go](https://github.com/rubenv/sql-migrate#writing-migrations), etc...) I did notice the separate-file characteristic of some migration tools, but I could not really understand the purpose. Could you share your use case with me? 

My reasoning for sticking with one file was:
  1. It hasn't caused me trouble yet.
  2. It seems fairly commonplace. 
  3. Programming is hard and just splitting the file contents on `---` was easy 😄 .

> In our minds this makes that file far less useful as it can't really be treated as a proper migration without parsing it first.

I don't know what you mean by "proper migration", could you elaborate on that?

**Comment by chrisnicola** [comment]
Sorry @thebearingedge it isn't a really big deal, but what I mean is that the files would be compatible with being directly passed to psql, instead of requiring text parsing.

**Comment by chrisnicola** [comment]
I think the reason that other migration implementations have up/down in the same file is that they are JS code, rather than SQL. Another library I looked at, I think for [massive](https://github.com/PDMLab/massive-migrate) did keep separate up and down files for .sql migrations.

**Comment by thebearingedge** [comment]
Ok, that all makes sense. The file format as it exists is definitely not `psql` friendly. There is also another package that really inspired `pg-bump` named [`careen`](https://github.com/programble/careen). In fact, `pg-bump` is virtually a subset of Curtis's tool. I believe that project actually supports split files.

**Comment by chrisnicola** [comment]
Ah I hadn't seen that one before. What was it about careen that didn't suit your needs?

**Comment by thebearingedge** [comment]
There were only a few problems I found while using it. 
1. The time of migration application is [used as the schema journal's primary key](https://github.com/programble/careen/blob/3e0057cb856b3867656616755f062528a8fd7b79/lib/client/postgresql.ts#L53). This actually breaks migrations run in a transaction 😢.
2. The database connection details had to be [known by the config file](https://github.com/programble/careen#postgresql-example), complicating deployment significantly.
3. The application _and_ reversion of migrations were unboundedly [appended to the schema journal](https://github.com/programble/careen/blob/3e0057cb856b3867656616755f062528a8fd7b79/lib/client/postgresql.ts#L63) and I don't really care about tracking when something got reverted. For now I feel that`down` scripts are only useful during development to iterate on their respective `up` scripts, never intended for production use.
